### ISSUE TITLE
Fatal Error: QAudioDeviceInfo

### ISSUE THREAD

**Issue by sysv0id** [Audio4Linux/Viper4Linux-GUI#2]
Encountered this error while compiling.

```
In file included from visualization/audiostreamengine.cpp:51:
visualization/audiostreamengine.h:57:10: fatal error: QAudioDeviceInfo: No such file or directory
   57 | #include <QAudioDeviceInfo>
      |          ^~~~~~~~~~~~~~~~~~

compilation terminated.
make: *** [Makefile:693: audiostreamengine.o] Error 1
```

**Comment by timschneeb** [comment]
You are probably missing the qt multimedia module. Install it on Ubuntu with:
```
sudo apt-get build-dep qtmultimedia5-dev
```

**Comment by sysv0id** [comment]
The command said it picked 'qtmultimedia-opensource-src' as source package instead of 'qtmultimedia5-dev'. Installed all the dependencies including qtmultimedia5-dev. Still the same error.  Do I have to install this first before V4L? 

I also tried the Legacy version of this GUI and I haven't encountered any issues.

**Comment by timschneeb** [comment]
Yes, qtmultimedia5-dev is a required dependency. 

I've pushed an updated makefile that might fix your issue, please pull the repo, run qmake again and rebuild the project.

**Comment by sysv0id** [comment]
Still the same. Here's the complete output from terminal:

```
jdotdarko@i-5559:/var/git$ sudo git clone https://github.com/Audio4Linux/Viper4Linux-GUI
Cloning into 'Viper4Linux-GUI'...
remote: Enumerating objects: 1813, done.
remote: Counting objects: 100% (1813/1813), done.
remote: Compressing objects: 100% (889/889), done.
remote: Total 1813 (delta 1193), reused 1525 (delta 912), pack-reused 0
Receiving objects: 100% (1813/1813), 2.06 MiB | 243.00 KiB/s, done.
Resolving deltas: 100% (1193/1193), done.
jdotdarko@i-5559:/var/git$ cd Viper4Linux-GUI/
jdotdarko@i-5559:/var/git/Viper4Linux-GUI$ sudo qmake
jdotdarko@i-5559:/var/git/Viper4Linux-GUI$ sudo make
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/convolver.ui -o ui_convolver.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/firstlaunchwizard.ui -o ui_firstlaunchwizard.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/importandroid.ui -o ui_importandroid.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/log.ui -o ui_log.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/menueditor.ui -o ui_menueditor.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/palettedlg.ui -o ui_palettedlg.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/preset.ui -o ui_preset.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/settings.ui -o ui_settings.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic dialog/statusfragment.ui -o ui_statusfragment.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic mainwindow.ui -o ui_mainwindow.h
g++ -c -m64 -pipe -Wno-old-style-cast -Wno-unused-function -O2 -Wall -W -D_REENTRANT -DDISABLE_LEVEL -DDISABLE_WAVEFORM -DSPECTRUM_ANALYSER_SEPARATE_THREAD -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_MULTIMEDIA_LIB -DQT_DBUS_LIB -DQT_SVG_LIB -DQT_XML_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtNetwork -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtXml -I/usr/include/qt4/QtSvg -I/usr/include/qt4/QtDBus -I/usr/include/qt4/QtMultimedia -I/usr/include/qt4 -Ivisualization/fftreal -I3rdparty/WAF -I. -I. -o audiostreamengine.o visualization/audiostreamengine.cpp
In file included from visualization/audiostreamengine.cpp:51:
visualization/audiostreamengine.h:57:10: fatal error: QAudioDeviceInfo: No such file or directory
   57 | #include <QAudioDeviceInfo>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:693: audiostreamengine.o] Error 1

```

**Comment by timschneeb** [comment]
Seems like you're trying to compile the GUI with Qt4, however unlike the legacy version this one requires at least Qt5, since it uses more modern library functions. Install and switch your default qt version to Qt5. The GUI is guaranteed to work with qt version 5.9.8, 5.13.2 and 5.14.1.

**Comment by sysv0id** [comment]
Thanks! I uninstalled every Qt package related then reinstalled build-essential, qtcreator, qt5-default, and qtmultimedia5-dev. 

Pulled the repo then ran qmake to another dead-end:
```
jdotdarko@i-5559:/var/git/Viper4Linux-GUI$ sudo qmake
/var/git/Viper4Linux-GUI/V4L_Frontend.pro:139: isEmpty(var) requires one argument.
/var/git/Viper4Linux-GUI/V4L_Frontend.pro:140: isEmpty(var) requires one argument.
Project ERROR: Unknown module(s) in QT: svg
```
So I installed libqt5svg5* and commented out those lines in V4L_Frontend.pro. No more issues after that. Compilation is all good. 

Btw, I'm a long time Viper fan! I have them on almost every Android devices that I had even on a Windows 10 which has a crappy Viper port. Lol. I hope this project lives long.
